presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -967,6 +1024,17 @@ public static Object invoke(ConnectorSession session, MethodHandle handle, List<
         }
     }
 
+    @VisibleForTesting
+    @NotNull
+    public static Object createFailureFunction(Type type, RuntimeException e)
+    {
+        checkNotNull(e, "Exception is null");
+        String failureInfo = JsonCodec.jsonCodec(FailureInfo.class).toJson(Failures.toFailure(e).toFailureInfo());
+        FunctionCall jsonParse = new FunctionCall(QualifiedName.of("json_parse"), ImmutableList.of(toExpression(Slices.utf8Slice(failureInfo), VARCHAR)));
+        Cast nullCast = new Cast(new NullLiteral(), type.toString());
+        return new FunctionCall(QualifiedName.of("fail"), ImmutableList.of(jsonParse, nullCast));
```

Comment:
Technically, it's still a parametric function :)

Please, add a TODO to remove that once we fix the type system. No function should return UNKNOWN, since it's not a valid type and is not really a type variable, either.